mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -1380,6 +1380,27 @@ void expungeShouldCallAllPreDeletionHooks() throws Exception {
                     }));
             }
 
+            @Test
+            void deleteMailboxShouldCallAllPreDeletionHooks() throws Exception {
```

Comment:
Do we need test with `deleteSharedMailbox` case?

Reply:
IMO this concern of integration between feature arise when you are testing the vault hook + the all feature integration test.